cpp/dolfinx/mesh/graphbuild.cpp

```diff
@@ -364,42 +364,61 @@ mesh::build_local_dual_graph(CellType celltype,
             0, std::vector<std::int32_t>()};
   }
 
-  int tdim = mesh::cell_dim(celltype);
-  int num_cell_vertices = mesh::cell_num_entities(celltype, 0);
-  const std::int32_t num_cells = cells.size() / num_cell_vertices;
-
-  // Note: only meshes with a single cell type are supported
-  const graph::AdjacencyList<int> cell_facets
-      = mesh::get_entity_vertices(celltype, tdim - 1);
+  assert(cells.size() == celltypes.size());
 
-  // Determine maximum number of vertices for facet
+  // Create indexing offset for each cell type
+  // and determine max number of vertices per facet
+  std::vector<std::int32_t> cell_offsets = {0};
```

Comment:
Add `cell_offsets.reserve(cells.size() + 1);`

Reply:
Not really important, since `cells.size()` is tiny.